tests/profiles-enabled/test_shipments.py

```diff
@@ -1800,3 +1803,204 @@ def test_shipment_set_shadow(self):
+        url = iot_url
+        for endpoint in self.iot_enpoints:
+            if Url(iot_url) == Url(endpoint):
+                url = endpoint
+                break
+        return mocked_rpc_response(self.map_responses[url])
+
+    def url_encode_params(self, owner_id, next_token='', box='', active=None):
+        param_dict = dict(
+            active=active if active is not None else '',
+            ownerId=owner_id,
+            maxResults=test_settings.IOT_DEVICES_PAGE_SIZE,
+            in_bbox=box,
+            nextToken=next_token
+        )
+        return urlencode(param_dict)
+
+    @mock_iot
+    def test_get_devices_locations(self):
+
+        self.set_user(self.user_1)
+
+        url = reverse('devices-status', kwargs={'version': 'v1'})
+
+        # The first called url doesn't have the nextToken value
+        self.map_responses[self.iot_enpoints[0]] = self.iot_responses(OWNER_ID)
+
+        with mock.patch('apps.iot_client.requests.Session.get') as mock_get:
+            mock_get.side_effect = self.side_effects
+
+            response = self.client.get(url)
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            data = response.json()
+            self.assertEqual(len(data['data']), 5)
+            # Only one Api call is made to IOT_AWS_HOST
+            mock_get.assert_called_once()
+
+            # The response for first called url do have the nextToken value,
+            # there should be a second call to IOT_AWS_HOST
+            mock_get.reset_mock()
+            self.map_responses = {
+                self.iot_enpoints[0]: self.iot_responses(OWNER_ID, next_token=True),
+                self.iot_enpoints[3]: self.iot_responses(OWNER_ID)
+            }
+
+            response = self.client.get(url)
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            data = response.json()['data']
+            self.assertEqual(len(data), 2 * test_settings.IOT_DEVICES_PAGE_SIZE)
+            assert mock_get.call_count == 2
+
+            # ----------------------- active param validation ------------------------- #
+            self.map_responses[self.iot_enpoints[1]] = self.iot_responses(OWNER_ID, active=True)
+            self.map_responses[self.iot_enpoints[2]] = self.iot_responses(OWNER_ID, active=False)
+
+            # Lower case boolean value should succeed with 200 status code
+            active_url = url + '?active=true'
+            response = self.client.get(active_url)
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+            # Upper case boolean value should succeed with 200 status code
+            inactive_url = url + '?active=FALSE'
+            response = self.client.get(inactive_url)
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+            # A request with a query params other than true or false should fail with status code 400
+            bad_param_url = url + '?active=falso'
```

Comment:
maybe change it to something more obviously not true/false, to make it easier upon cursory review and if any other changes need to be made. Maybe something like NONBOOLEAN to make it very transparent.